### ISSUE TITLE
Fix anonymous type alias comparison

### ISSUE THREAD

**Issue by aitorres** [aitorres/firelink#137]
Anonymous type aliases are presenting errors, on parsing (case: passing an anonymous struct to a function). It could also be checked if two anonymous structures that are semantically the same type, are being recognized as the same type too.

**Comment by aitorres** [comment]
Testing with

```
hello ashen one

invocation f
requesting
    val x of type bezel { a of type humanity, b of type hollow }
with skill of type humanity
    traveling somewhere
        go back with x~>a
    you died
after this return to your world

traveling somewhere
with
    var x of type bezel { a of type humanity, b of type hollow }
in your inventory
    x~>a <<= 1 \
    x~>b <<= 9.3
you died

farewell ashen one
```

**Comment by aitorres** [comment]
Closed by 36686466239d45ffc206fbadbede3e359d78ce11
